blob/fileblob/fileblob_test.go

```diff
@@ -33,27 +34,62 @@ import (
 )
 
 type harness struct {
-	dir    string
-	closer func()
+	dir        string
+	server     *httptest.Server
+	urlSignKey string
+	closer     func()
 }
 
 func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
 	dir := path.Join(os.TempDir(), "go-cloud-fileblob")
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return nil, err
 	}
-	return &harness{
-		dir:    dir,
-		closer: func() { _ = os.RemoveAll(dir) },
-	}, nil
+
+	h := &harness{dir: dir, urlSignKey: "I'm a secret key"}
+	localServer := httptest.NewServer(http.HandlerFunc(h.serveSignedURL))
+	h.server = localServer
+	h.closer = func() { _ = os.RemoveAll(dir); localServer.Close() }
+
+	return h, nil
+}
+
+func (h *harness) serveSignedURL(w http.ResponseWriter, r *http.Request) {
+	// URLSignerHMAC requires the scheme, host, and path to construct the
+	// signed URL, but validating only requires the key.
+	urlSigner := NewURLSignerHMAC("", "", "", h.urlSignKey)
+	objKey, ok := urlSigner.KeyFromURL(r.Context(), r.URL)
+	if !ok {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
+	bucket, err := OpenBucket(h.dir, &Options{})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	blobject, err := bucket.ReadAll(r.Context(), objKey)
```

Comment:
(Doesn't need to be addressed for this PR, but thought I'd mention it for when this handler is exported. Perhaps add a TODO.)

This access method would load the entire blob into memory and then write it. A more efficient way to do this would be to use `blob.NewReader` to get a reader and then use `io.Copy` to copy from the reader to the `http.ResponseWriter`.